### PR TITLE
[4.0] Blog layout ux

### PR DESCRIPTION
**Pull request by chmst**
Pull Request for Issue #32012 .

### Summary of Changes
J4 uses css grid classes for the multi column presentation of blogs. This is done via modifying classes as described here: https://github.com/joomla/joomla-cms/pull/18319.

For users, espcially for users who migrate from J3 this is not self explaining, they ares used do define a fixed number of columns and display diretion. 

This PR enables both, using own classes - but also defining a number of columns and a output direction. 
This PR uses classes from #32488 for the blog layout tab where the blog lyout is definedt. This should be self explaining 

It accepts columns and ordering for blog layouts in two extra fields. The input in these fields is converted to css grid classes columns-x or mansonry-x in the respective layouts. Please see comment https://github.com/joomla/joomla-cms/pull/32943#issuecomment-813485654
grafik

If this is accepted, the migration script from j3.10 to 4.0 ist no longer necessary.

### Testing Instructions

Testing of this step is easy, just apply the patch, set your global params in content options. A
Add new menu Items to category blogs, play around with settings, or change existing menuItems of this type.

### Documentation Changes Required
yes

### CODE REVIEW

File: administrator/language/en-GB/joomla.ini [hunk 1]
```diff
@@ -354,9 +355,14 @@ JGLOBAL_BATCH_MOVE_ROW_NOT_FOUND="Can't find the destination row for this move."
...
 JGLOBAL_BLOG="Blog"
-JGLOBAL_BLOG_CLASS="Blog Class"
-JGLOBAL_BLOG_CLASS_LEADING="Blog Class (Leading Articles)"
+JGLOBAL_BLOG_CLASS="Article Class"
+JGLOBAL_BLOG_CLASS_LEADING="Leading Article Class"
 JGLOBAL_BLOG_LAYOUT_OPTIONS="Blog Layout"
+JGLOBAL_BLOG_STYLE="Blog Layout Style"
+JGLOBAL_BLOG_STYLE_J3="Joomla 3 style"
+JGLOBAL_BLOG_STYLE_J4="Joomla 4 style: Grid Layout"
+JGLOBAL_BLOG_STYLE_J4_DESC="You can add every css class for your own styling ideas. <br>Generate columns with classes column-2, column-3 and so on <br><br>Add a border with class boxed.<br>For example: column-3 boxed will make you athree columnlyout with border in your primary template colour."
```
Comment (brianteeman) [inline administrator/language/en-GB/joomla.ini]:
```suggestion
JGLOBAL_BLOG_STYLE_J4_DESC="You can add any css class for your own styling ideas. <br>Generate columns with classes column-2, column-3 etc.<br><br>Add a border with class boxed.<br>For example: column-3 boxed will make you a three column layout with a border in your primary template colour."
```

**Comment by brianteeman** [comment]
I have had a super quick look before jumping to a day long meeting. Looks great!!!!

One suggestion would be to use different terminology to J3 and J4. A user might be new to joomla and so it wont meaning anything to them.

Not sure what to call the J3 version - perhaps "fixed count"
For the J4 how about "CSS grid"

**Comment by brianteeman** [comment]
Please add the release blocker label

**Comment by richard67** [comment]
A question which is related to this PR and might have to be handled with another, new PR is the handling of updates from 3.10.

With PR #31570 , a migration of the old J3 blog layout parameters to the new css classes on update has been implemented.

When this PR here will be merged, it might sense to think about reverting it so people keep their old J3 values on update, and having the Joomla 3 way as default (params value in db) for the new created field here, while for new installations the default would be the Joomla 4 way.

**Comment by chmst** [comment]
@richard67 thanks, your link is the correct one. 
The script should add the param blog_style = 3 and let other params untouched.

**Comment by brianteeman** [comment]
Why do we even need the script any more

/me always reluctant to touch user data

**Comment by richard67** [comment]
> The script should add the param blog_style = 3 and let other params untouched.

@chmst You mean when updating from 3.10 leave other params untouched? This would mean to revert #31570 almost completely. When updating previous 4.0 Beta versions, the blog classes created by the conversion from that PR would be not touched. Or did you mean let other params untouched compared to what we have now in 4.0-dev without this PR? I think you meant the 1st way, updating from 3.10, but as I'm not sure if I understand right, I'm asking.

File: administrator/language/en-GB/joomla.ini [hunk 2]
```diff
@@ -354,9 +355,14 @@ JGLOBAL_BATCH_MOVE_ROW_NOT_FOUND="Can't find the destination row for this move."
...
 JGLOBAL_BATCH_WORKFLOW_STATE_ROW_NOT_FOUND="Can't find the destination row for this state change."
 JGLOBAL_BLOG="Blog"
-JGLOBAL_BLOG_CLASS="Blog Class"
-JGLOBAL_BLOG_CLASS_LEADING="Blog Class (Leading Articles)"
+JGLOBAL_BLOG_CLASS="Article Class"
+JGLOBAL_BLOG_CLASS_LEADING="Leading Article Class"
 JGLOBAL_BLOG_LAYOUT_OPTIONS="Blog Layout"
+JGLOBAL_BLOG_STYLE="Blog Layout Style"
+JGLOBAL_BLOG_STYLE_J3="Joomla 3 style (Column Count)"
+JGLOBAL_BLOG_STYLE_J4="Joomla 4 style ( CSS Grid)"
```
Comment (Quy) [inline administrator/language/en-GB/joomla.ini]:
```suggestion
JGLOBAL_BLOG_STYLE_J4="Joomla 4 style (CSS Grid)"
```

File: administrator/language/en-GB/joomla.ini [hunk 3]
```diff
@@ -302,6 +302,7 @@ JFIELD_VERSION_HISTORY_DESC="This button allows you to open a window to view old
 JFIELD_VERSION_HISTORY_LABEL="Prior Versions"
 JFIELD_VERSION_HISTORY_SELECT="View Prior Versions"
 JGLOBAL_ACTION_PERMISSIONS_LABEL="Permissions"
+JGLOBAL_ACROSS="Across"
```
Comment (Quy) [inline administrator/language/en-GB/joomla.ini]:
```suggestion
JGLOBAL_ACROSS="Across"
JGLOBAL_ACTION_PERMISSIONS_LABEL="Permissions"
```

**Comment by chmst** [comment]
I see two cases: In both of them, the script is used but simplified:

From 3.10 to 4: I would use the script und add the param "blog_style":"3" to params

From 4.0 beta to 4.0: I would use the script und add the param "blog_style":"4" to params


<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/32943">issues.joomla.org/tracker/joomla-cms/32943</a>.</sub>

**Comment by brianteeman** [comment]
> From 3.10 to 4: I would use the script und add the param "blog_style":"3" to params

Cant it be 3 by default? Then you dont need to change anything

Re the name I really am not in favour of anything be labelled "Joomla 3 style" Firstly the user may never have used Joomla before so it is a meaningless term to them. Secondly if you wanted to be really accurate then it would be "Joomla 1 style"

File: administrator/language/en-GB/joomla.ini (hunk 3 above)
Comment (chmst) [reply administrator/language/en-GB/joomla.ini]: other solution in last commit

File: administrator/language/en-GB/joomla.ini (hunk 2 above)
Comment (chmst) [reply administrator/language/en-GB/joomla.ini]: changed in last commit

File: administrator/language/en-GB/joomla.ini [hunk 4]
```diff
@@ -354,9 +354,16 @@ JGLOBAL_BATCH_MOVE_ROW_NOT_FOUND="Can't find the destination row for this move."
...
 JGLOBAL_BLOG="Blog"
-JGLOBAL_BLOG_CLASS="Blog Class"
-JGLOBAL_BLOG_CLASS_LEADING="Blog Class (Leading Articles)"
+JGLOBAL_BLOG_ACROSS_OPTION="Across"
+JGLOBAL_BLOG_DOWN_OPTION="Down"
+JGLOBAL_BLOG_CLASS="Article Class"
+JGLOBAL_BLOG_CLASS_LEADING="Leading Article Class"
 JGLOBAL_BLOG_LAYOUT_OPTIONS="Blog Layout"
+JGLOBAL_BLOG_STYLE="Blog Layout Style"
+JGLOBAL_BLOG_STYLE_FIXED_OPTION="Fixed Columns and direction"
```
Comment (richard67) [inline administrator/language/en-GB/joomla.ini]: Inconsistent capitalisation. Maybe @brianteeman can advise how it's right. I think it should be "Fixed Columns and Direction", but I might be wrong.

File: administrator/language/en-GB/joomla.ini [hunk 5]
```diff
@@ -354,9 +354,16 @@ JGLOBAL_BATCH_MOVE_ROW_NOT_FOUND="Can't find the destination row for this move."
...
+JGLOBAL_BLOG_STYLE_GRID_OPTION="CSS Grid with modifier classes"
```
Comment (richard67) [inline administrator/language/en-GB/joomla.ini]: Same as above: Why "Grid" capitalised and the rest after it not?

File: administrator/language/en-GB/joomla.ini [hunk 6]
```diff
@@ -354,9 +354,16 @@ JGLOBAL_BATCH_MOVE_ROW_NOT_FOUND="Can't find the destination row for this move."
...
+JGLOBAL_BLOG_STYLE_GRID_DESC="You can add any css class for your own styling ideas. <br>Generate columns with classes column-2, column-3 etc.<br><br>Add a border with class boxed.<br>For example: column-3 boxed will make you a three column layout with a border in your primary template colour."
+JGLOBAL_BLOG_STYLE_GRID_LABEL="How to use modifier classes for your layout:"
```
Comment (richard67) [inline administrator/language/en-GB/joomla.ini]: Normally our labels are capitalised, but I'm not sure if this also applies to such a long label like here. Maybe Brian can advise?

Comment (chmst) [reply administrator/language/en-GB/joomla.ini]: In this case it is a label in the field, but for the user it is a description. But I let the decision - generally for texts, to native speakers

Comment (brianteeman) [reply administrator/language/en-GB/joomla.ini]: The issue is that it shouldnt be a spacer field but a note field. I have submitted a PR to your branch with that change and updated the string constants etc etc https://github.com/chmst/joomla-cms/pull/11

Comment (brianteeman) [reply administrator/language/en-GB/joomla.ini]: PS - thanks for changing the option names!!

File: administrator/language/en-GB/joomla.ini [hunk 7]
```diff
@@ -511,6 +518,7 @@ JGLOBAL_MODIFIED="Modified"
 JGLOBAL_MODIFIED_DATE="Modified Date"
 JGLOBAL_MOST_HITS="Most Hits"
 JGLOBAL_MOST_RECENT_FIRST="Most Recent First"
+JGLOBAL_MULTI_COLUMN_ORDER_LABEL="Multi Column Order"
```
Comment (brianteeman) [inline administrator/language/en-GB/joomla.ini]: I know this is what we had before but as it is now referred to as "Fixed Columns and Direction" I think we should change this to 
```suggestion
JGLOBAL_MULTI_COLUMN_ORDER_LABEL="Multi Column Direction"
```

**Comment by ceford** [comment]
For testing I created a category Monarchs and six lorem ipsum articles with Readmore partings and a Category Blog menu item. The Cassiopeia template shows five items: always in the order created, always in one column, always with the boxed style if that has been set and never with links to older articles. The column-2 style seems to do nothing. I don't do much production work with Articles so I may have missed some important step in testing. I did a fresh Pull and Merge and npm ci before testing. I see the test instructions say `In cassiopeia or other J4 templates nothing changes in the frontend.` which makes all this seems very confusing.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/32943">issues.joomla.org/tracker/joomla-cms/32943</a>.</sub>

**Comment by brianteeman** [comment]
Exact same findings as @ceford

**Comment by frostmakk** [comment]
> The column-2 style seems to do nothing.

The class is plural form columns-2
I strongly believe there should be some kind of lookup to aid the user here.
Currently available classes like masonry-3, image-left, image-right, image-bottom & image-alternate are still among the "secret" classes.

**Comment by chmst** [comment]
@ceford for this PR no npm is needed, it was a typo in the description. With for example "columns-2 boxed" you will get 2 columns.
Thank you @frostmark, I changed the description. <hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/32943">issues.joomla.org/tracker/joomla-cms/32943</a>.</sub>

**Comment by astridx** [comment]
I have tested the PR. And it all worked. Here are a few thoughts.

- Is it possible to separate the fields that are shown and hidden by `showon`. For newbies, and that are almost all when switching to Joomla 4, the orientation would be easier. A little more space would be helpful.

Menus Edit Item TestBlogLayout Administration(1)

- Is it possible to show a hint with the option _Fixed Columns and Direction_ value as well? If I change the number of columns here, this does not work in every template - also in Cassiopeia. A hint that this depends on the template would be helpful. Especially because it doesn't work in Cassiopeia. Or am i doing something wrong?

- I see it correctly that the classes of the option _CSS Grid with Modifier Classes_ are set in the `template.css` of Cassiopeia, right? A template developer may override the option, but if not: Add the info, that it depends on the template in this hint, too. 

- Is it possible to list the classes with the option _CSS Grid with Modifier Classes_  , there are not so many (https://github.com/joomla/joomla-cms/pull/18319). There are certainly those who test `boxed columns-5` and then are unsure, because only one column is shown. Or write it less detailed and put a reference to the help button in the upper area. Because the `#` in `Generate columns with class columns-# or masonry-#.` indicates for me, that I can use what I want.

**Comment by chmst** [comment]
@Astrid thank you - I am on it. 

My intention was to enable a migration so that templates after a migration from 3.10 to j4 can be used without changes. It is the same in J3.x, that it depends on the template if multicolumns or the direction is used and cassiopeia does not support it. 
I can use the num_columns and transform it to columns-x. But really don't know if this makes sense. 
<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/32943">issues.joomla.org/tracker/joomla-cms/32943</a>.</sub>

File: administrator/language/en-GB/joomla.ini (hunk 6 above)
Comment (brianteeman) [reply administrator/language/en-GB/joomla.ini]: Woould be great if you could merge or comment on my pr

**Comment by brianteeman** [comment]
For me it is still a release blocker if both sets of options do not work with the default template

**Comment by chmst** [comment]
I have tried  added more description. If this is understandable, we can add an override in cassiopeia so that both layouts grid and num_columns can be used in cassiopeia.
<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/32943">issues.joomla.org/tracker/joomla-cms/32943</a>.</sub>

File: components/com_content/tmpl/category/blog.xml [hunk 8]
```diff
@@ -37,317 +37,384 @@
...
+				type="spacer"
+				parentclass="span-6"
+			/>
+
+			<field
+				name="blog_information"
+				type="spacer"
+				label="JGLOBAL_BLOG_STYLE_LABEL"
+				parentclass="stack span-2"
+				description="JGLOBAL_BLOG_STYLE_DESC"
```
Comment (Quy) [inline components/com_content/tmpl/category/blog.xml]:
```suggestion
				label="JGLOBAL_BLOG_STYLE_LABEL"
				description="JGLOBAL_BLOG_STYLE_DESC"
				parentclass="stack span-2"
```

File: components/com_content/tmpl/category/blog.xml [hunk 9]
```diff
@@ -37,317 +37,384 @@
...
+				<option value="0">JGLOBAL_BLOG_STYLE_FIXED_OPTION</option>
+				<option value="1">JGLOBAL_BLOG_STYLE_GRID_OPTION</option>
+			</field>
+
+			<field
+				name="blog_information-grid"
+				type="spacer"
+				label="JGLOBAL_BLOG_STYLE_GRID_LABEL"
+				parentclass="stack span-2"
+				description="JGLOBAL_BLOG_STYLE_GRID_DESC"
```
Comment (Quy) [inline components/com_content/tmpl/category/blog.xml]:
```suggestion
				label="JGLOBAL_BLOG_STYLE_GRID_LABEL"
				description="JGLOBAL_BLOG_STYLE_GRID_DESC"
				parentclass="stack span-2"
```

File: components/com_content/tmpl/category/blog.xml [hunk 10]
```diff
@@ -37,317 +37,384 @@
...
+				validate="CssIdentifier"
+				showon="blog_style:1"
+			/>
+
+			<field
+				name="blog_information-fixed"
+				type="spacer"
+				label="JGLOBAL_BLOG_STYLE_FIXED_LABEL"
+				parentclass="stack span-2"
+				description="JGLOBAL_BLOG_STYLE_FIXED_DESC"
```
Comment (Quy) [inline components/com_content/tmpl/category/blog.xml]:
```suggestion
				label="JGLOBAL_BLOG_STYLE_FIXED_LABEL"
				description="JGLOBAL_BLOG_STYLE_FIXED_DESC"
				parentclass="stack span-2"
```

**Comment by brianteeman** [comment]
I can see what you mean with the text. It needs a little "english tweaking" which I can do if you want. 
No idea why you are ignoring my pr to change the spacer to a note :(

**Comment by Harmageddon** [comment]
Concerning the upload script introduced in #31570: If this PR is merged, there is no need anymore to convert the num_columns to blog classes during 3->4 upgrade. In this case, we can revert most of #31570. The only remaining distinction is the question which style should be enabled during upgrade. I agree with @chmst here:

> I see two cases: In both of them, the script is used but simplified:
> 
> From 3.10 to 4: I would use the script und add the param "blog_style":"3" to params
> 
> From 4.0 beta to 4.0: I would use the script und add the param "blog_style":"4" to params

And as suggested by @brianteeman, we can even get rid of the first case if we set the blog style to "fixed" by default. If we go for the CSS grid style as default, we get rid of the second case instead. In either case, I would provide a PR changing the update script accordingly after this one here is merged.

(Nothing really new to the discussion here, I just wanted to summarize the above discussion about updating/migrating, to clarify that the changes introduced with this PR here can easily be included in the update script.)

**Comment by chmst** [comment]
@brianteeman my own repo did not send notofications, so did not see your PR, sorry for that. And thank you for the texts.

**Comment by chmst** [comment]
I have spent some hours yesterday, writing overrides for cassiopeia using the new param blog_style. 

But the more time I spent with these params the more I  feel that it is wrong. "Don't Make me Think".  But  this makes me think.

I want to add class like this  "myownclass boxed columns-2 image-bottom" to the blog. We use the css grid layout and cassiopeia as prepared / overridden someclasses for it (for example .boxed a,d classes for images)
https://github.com/joomla/joomla-cms/blob/4.0-dev/templates/cassiopeia/scss/blocks/_modifiers.scss#L83

Expereinced users can enter this in the input field blog_class as they want. 

Unexperienced users or users who come from J3 are better served with the possibility to define a number of columns and the order of the articles in the columns. This is possible now, but the param blog_style introduced in this PR is not really self explaining and not really helpful. 

So why not simple accept both: A Free definded classes for everything and num columns and ordering? 

The advantage:
- Migration from J3 is possible without changes, we need no script at all 
- Users can classes from cassiopiea (i.e. boxed) or custom classes
- No new parameter which probably will not be used
- In the layout, it is very simple to make columns-x or masonry-x from the two number fields. 

It could look like here: 

grafik

**Comment by brianteeman** [comment]
@chmst assuming that I understand you correctly that is perfect and what we've wanted since the beginning.. For whatever reason we were told that this was wrong and that it had to be classes only - hence the release blocker and the "compromise" you originally proposed.

**Comment by frostmakk** [comment]
What's with all this migration talk?
The purpose of this form should be to present the users with all the options available, and how to use them, in the most self explanatory way possible, to enable them to create the blog they want.
When that purpose is fulfilled, the number of columns in the old template becomes more or less irrelevant, and can be set to default values. When the users discover how to control the CSS grid with the classes provided to them, they will experiment for hours, and it will be one of the most positive experiences they get from J4.

**Comment by brianteeman** [comment]
> they will experiment for hours,

If they have to do that then Joomla is dead in the water

**Comment by chmst** [comment]
@frostmakk experienced users can do that. They can experiment for hours, we don't cut anything with these changes. 
But for unexperienced users it hopefully is easier now.

**Comment by frostmakk** [comment]
> If they have to do that then Joomla is dead in the water 


They don't have to.
They will do it in pure joy over the powers given to them.

**Comment by chmst** [comment]
With the last changes, user can define their classes free. 

Either in the class field as for example "boxed image-right masonry-3"
Or as in J3 enter a number of columns and a ordering. 

What now could happen: A user writes class "column-3" into the class field and then sets num_columns to 2. This would result in a class like "column-3 column-2".  No idea how to avoid this.

**Comment by ChristineWk** [comment]
> What now could happen: A user writes class "column-3" into the class field and then sets num_columns to 2. This would result in a class like "column-3 column-2". No idea how to avoid this.

Hmm :-) A note: class cannot (or should) be named in the same way as column numbers

**Comment by richard67** [comment]
> What now could happen: A user writes class "column-3" into the class field and then sets num_columns to 2. This would result in a class like "column-3 column-2". No idea how to avoid this.

@chmst We could use the average, "column-2.5" ;-) (forgive me my silly joke)

File: components/com_content/tmpl/category/blog.xml [hunk 11]
```diff
@@ -667,7 +683,7 @@
 			</field>
 
 			<field
-				name="feed_summary"
+				name="feed_"
```
Comment (Quy) [inline components/com_content/tmpl/category/blog.xml]: Remove `_`?

File: administrator/components/com_content/config.xml [hunk 12]
```diff
@@ -717,47 +717,69 @@
...
-			name="num_leading_articles"
-			type="number"
-			label="JGLOBAL_NUM_LEADING_ARTICLES_LABEL"
+			name="num_columns"
+			type="list"
+			label="JGLOBAL_NUM_COLUMNS_LABEL"
 			default="1"
-		/>
+			parentclass="stack span-1-inline"
+		>
```
Comment (Quy) [inline administrator/components/com_content/config.xml]:
```suggestion
			>
```

File: components/com_content/tmpl/category/blog.xml [hunk 13]
```diff
@@ -37,317 +37,333 @@
...
+				label="JGLOBAL_NUM_LINKS_LABEL"
+				parentclass="stack span-1"
+				useglobal="true"
+			/>
+
+			<field
+				name="show_featured"
+				type="list"
+				default=""
+				label="JGLOBAL_SHOW_FEATURED_ARTICLES_LABEL"
```
Comment (Quy) [inline components/com_content/tmpl/category/blog.xml]:
```suggestion
				label="JGLOBAL_SHOW_FEATURED_ARTICLES_LABEL"
				default=""
```

File: components/com_content/tmpl/category/blog.xml (hunk 11 above)
Comment (chmst) [reply components/com_content/tmpl/category/blog.xml]: fixed, thanks,

File: administrator/language/en-GB/joomla.ini (hunk 6 above)
Comment (chmst) [reply administrator/language/en-GB/joomla.ini]: JGLOBAL_BLOG_CLASS_NOTE_DESC added.

File: administrator/language/en-GB/joomla.ini (hunk 7 above)
Comment (chmst) [reply administrator/language/en-GB/joomla.ini]: done

**Comment by michwebdev** [comment]
I have tested this item :white_check_mark: successfully on d236a2e3e19c6dabeaec2681446bb81eb0f03310<br /><br />What I observed is that the patch makes changes to a blog menu item configuration in the Blog Layout editing tab.  The patched version is a more intuitive layout.  

One issue I noticed with the layout is that when the screen width is under 1450 px, the "Multi Column Direction" and "Date for Ordering" fields extend outside of the border around the editing area.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/32943">issues.joomla.org/tracker/joomla-cms/32943</a>.</sub>

**Comment by BrentCritch** [comment]
I have tested this item :white_check_mark: successfully on d236a2e3e19c6dabeaec2681446bb81eb0f03310<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/32943">issues.joomla.org/tracker/joomla-cms/32943</a>.</sub>

**Comment by richard67** [comment]
> One issue I noticed with the layout is that when the screen width is under 1450 px, the "Multi Column Direction" and "Date for Ordering" fields extend outside of the border around the editing area.

That's a know issue, we are working on it.

**Comment by richard67** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/32943">issues.joomla.org/tracker/joomla-cms/32943</a>.</sub>

**Comment by cdguillaudeu** [comment]
I have tested this item :white_check_mark: successfully on d236a2e3e19c6dabeaec2681446bb81eb0f03310<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/32943">issues.joomla.org/tracker/joomla-cms/32943</a>.</sub>

File: administrator/language/en-GB/joomla.ini [hunk 14]
```diff
@@ -354,8 +354,11 @@ JGLOBAL_BATCH_MOVE_ROW_NOT_FOUND="Can't find the destination row for this move."
 JGLOBAL_BATCH_PROCESS="Process"
 JGLOBAL_BATCH_WORKFLOW_STATE_ROW_NOT_FOUND="Can't find the destination row for this state change."
 JGLOBAL_BLOG="Blog"
-JGLOBAL_BLOG_CLASS="Blog Class"
-JGLOBAL_BLOG_CLASS_LEADING="Blog Class (Leading Articles)"
+JGLOBAL_BLOG_ACROSS_OPTION="Across"
+JGLOBAL_BLOG_CLASS="Article Class"
+JGLOBAL_BLOG_CLASS_LEADING="Leading Article Class"
+JGLOBAL_BLOG_CLASS_NOTE_DESC ="You can add any css class for your own styling ideas.<br>Add a border on top with class boxed<br>For image position use for example image-left, image-right. Add image-alternate for alternative ordering of intro-images."
```
Comment (Quy) [inline administrator/language/en-GB/joomla.ini]:
```suggestion
JGLOBAL_BLOG_CLASS_NOTE_DESC ="You can add any CSS class for your own styling ideas.<br>Add a border on top with class boxed.<br>For image position use for example image-left, image-right. Add image-alternate for alternative ordering of intro-images."
```

**Comment by frostmakk** [comment]
Another wasted opportunity to give new users the wow experience that Joomla needs right now.
CSS grid is a novelty in J, and classes are the new toys to play with.
Give the users a proper interface with classes to choose from, not this unclear halfway something solution.
This is just sad.

**Comment by richard67** [comment]
> Another wasted opportunity to give new users the wow experience that Joomla needs right now.
> CSS grid is a novelty in J, and classes are the new toys to play with.
> Give the users a proper interface with classes to choose from, not this unclear halfway something solution.
> This is just sad.

@frostmakk You are just ignoring here the answers you have already been given in previous comments. This PR removes nothing of the flexibility you have for that which is already there in J4, it just offers an additional possibility to do it also in the other way which many people seem to prefer.

**Comment by N6REJ** [comment]
> > If they have to do that then Joomla is dead in the water
> 
> They don't have to.
> They will do it in pure joy over the powers given to them.

not at all!  I can tell you as a VERY experienced person that the minute you start trying to "make life simple" for me, you hinder me greatly and make my job 100x harder.
A simple example is helix.  powerful UI.. nightmare for me to "customize" to the degree that I'm paid for from within the ui itself.

**Comment by Quiviro** [comment]
I have tested this item :white_check_mark: successfully on 4ef5eee1a910446339aa5151923ecea2bcf212bb<br /><br />I changed the content options of the menu item and it reflects in the blog layout<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/32943">issues.joomla.org/tracker/joomla-cms/32943</a>.</sub>

**Comment by TLWebdesign** [comment]
I have tested this item :white_check_mark: successfully on 4ef5eee1a910446339aa5151923ecea2bcf212bb<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/32943">issues.joomla.org/tracker/joomla-cms/32943</a>.</sub>

**Comment by frostmakk** [comment]
Is my English really that bad?
None of You have understood what I am trying to communicate.

**Comment by chmst** [comment]
@frostmakk maybe my understanding of English is so bad. This is what I understand.
You would like the opportunity to use the full concept of css grid classes for the blog Layout. And you think that our layout should prepare all possibilities in a way that they must not know the grid concept or classes.

**Comment by frostmakk** [comment]
I would like the users to be able to do a lookup of all the available classes in the template, and select the combinations they want.
The lookup and selection could be a separate popup window or whatever you call a dialogue form.
The input field should then be populated with the selection.
Experienced users can put whatever they want in the input fields without using the lookup.

**Comment by frostmakk** [comment]
Sacrificed on the Release Blocker altar.

**Pull request merged**